File: sm-resources/src/main/java/com/bkk/sm/authentication/model/UserResource.java
package com.bkk.sm.authentication.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserResource {
    private String id;
    private String username;
    private String password;
    private String firstName;
    private String lastName;
    private String middleName;
    private String email;
    private List<CompanyRoleResource> roles;
    private Integer failedLoginAttempts = 0;
    int version;
    private Date registrationTime;
    private Date lastModificationTime;
    private Date passwordExpiryTime;
    private Date activatedTime;
    private boolean accountExpired = false;
    private boolean accountLocked = false;
    private boolean enabled = false;
    private String activationCode;
}

File: sm-common-models/src/main/kotlin/com/bkk/sm/common/customer/resources/UserResource.kt
package com.bkk.sm.common.customer.resources

import com.bkk.sm.common.customer.company.CompanyRole
import java.time.ZonedDateTime

data class UserResource(
    var id: String? = null,
    var username: String,
    var password: String? = null,
    var firstName: String = "",
    var lastName: String = "",
    var email: String = "",
    var failedLoginAttempts: Int? = 0,
    var roles: MutableList<CompanyRole>? = ArrayList(),
    var registrationTime: ZonedDateTime? = null,
    var lastModificationTime: ZonedDateTime? = null,
    var passwordExpiringEnabled: Boolean = true,
    var passwordExpiryTime: ZonedDateTime? = null,
    var activationKey: String? = null,
    var activatedTime: ZonedDateTime? = null,
    var accountLocked: Boolean = true,
    var enabled: Boolean = false,
    var middleName: String? = null,
)
